integrations/gcp/src/commonMain/kotlin/com/xebia/functional/xef/gcp/GCP.kt

```diff
@@ -0,0 +1,40 @@
+package com.xebia.functional.xef.gcp
+
+import arrow.core.nonEmptyListOf
+import com.xebia.functional.xef.AIError
+import com.xebia.functional.xef.conversation.Conversation
+import com.xebia.functional.xef.env.getenv
+import com.xebia.functional.xef.store.LocalVectorStore
+import com.xebia.functional.xef.store.VectorStore
+import kotlin.jvm.JvmField
+
+private const val GCP_TOKEN_ENV_VAR = "GCP_TOKEN"
+
+class GCP(projectId: String, location: VertexAIRegion, token: String? = null) {
+  private val config =
+    GcpConfig(
+      token = token ?: tokenFromEnv(),
+      projectId = projectId,
+      location = location,
+    )
+
+  private fun tokenFromEnv(): String =
+    getenv(GCP_TOKEN_ENV_VAR)
+      ?: throw AIError.Env.GCP(nonEmptyListOf("missing $GCP_TOKEN_ENV_VAR env var"))
+
+  val CODECHAT by lazy { GcpModel("codechat-bison@001", config) }
+  val TEXT_EMBEDDING_GECKO by lazy { GcpModel("textembedding-gecko", config) }
+
+  @JvmField val DEFAULT_CHAT = CODECHAT
+  @JvmField val DEFAULT_EMBEDDING = TEXT_EMBEDDING_GECKO
+
+  fun supportedModels(): List<GcpModel> = listOf(CODECHAT, TEXT_EMBEDDING_GECKO)
+}
+
+suspend inline fun <A> GCP.conversation(
```

Comment:
The `GPC` receiver parameter is never used. Maybe we can use `VectorStore` as the receiver, but for now, we are not using this function, I'm not sure if we can remove it

Reply:
Making VectorStore the receiver wouldn't align with the rest. Indeed, it may be removed.